src/java/org/apache/cassandra/db/marshal/AbstractType.java

```diff
@@ -117,21 +117,10 @@ protected AbstractType(ComparisonType comparisonType, boolean isMultiCell, List<
             throw new IllegalStateException();
         }
 
-        comparatorSet = new ValueComparators(new Comparator<>()
-        {
-            @Override
-            public int compare(byte[] l, byte[] r)
-            {
-                return AbstractType.this.compare(l, ByteArrayAccessor.instance, r, ByteArrayAccessor.instance);
-            }
-        }, new Comparator<>()
-        {
-            @Override
-            public int compare(ByteBuffer l, ByteBuffer r)
-            {
-                return AbstractType.this.compare(l, ByteBufferAccessor.instance, r, ByteBufferAccessor.instance);
-            }
-        });
+        comparatorSet = new ValueComparators((l, r) ->
```

Comment:
This is going to break JDK22

JAMM doesn't work wll with lamdas.


https://github.com/datastax/cassandra/commit/5a144533671fc6959532f4b8ac51d1f1df89ddd5